client/main.go

```diff
@@ -216,10 +227,21 @@ func parseStreamingRatio(streamingRatio string) (int64, int64) {
 }
 
 func sendStreamingRequests(client pb.ResponderClient,
-	shutdownChannel <-chan struct{}, lengthDistribution distribution.Distribution,
+	shutdownChannel <-chan struct{}, clientTimeout time.Duration,
+	lengthDistribution distribution.Distribution,
 	latencyDistribution distribution.Distribution, streamingRatio string,
 	r *rand.Rand, received chan *MeasuredResponse) error {
-	stream, err := client.StreamingGet(context.Background())
+
+	var ctx context.Context
+	if clientTimeout != time.Duration(0) {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(context.Background(), clientTimeout)
+		defer cancel()
+	} else {
+		ctx = context.Background()
+	}
+
+	stream, err := client.StreamingGet(ctx)
```

Comment:
Applying a timeout to the context here applies to the lifetime of the entire stream, rather than individual requests. So you would need to set `clientTimeout` higher than total test time in order for the test to not exit prematurely. My votes is to not apply `clientTimeout` to streaming requests at all, and just change the description of that flag to indicate that it's only applicable to non-streaming requests.